Sources/GRPCNIOTransportCore/GRPCStreamStateMachine.swift

```diff
@@ -1153,9 +1161,13 @@ extension GRPCStreamStateMachine {
       }
 
     case .clientOpenServerClosed, .clientClosedServerClosed:
-      try self.invalidState(
-        "Cannot have received anything from a closed server."
-      )
+      // If the server is closed, it's because it actually closed, or because the client
+      // transitioned it to a close state because it returned invalid headers.
+      // In either case, we will have already surfaced a status + trailers response to the client,
+      // so if we receive further packages, we should just drop them on the floor,
+      // as there's nothing for us to do with them.
+      return .doNothing
```

Comment:
There are presumably other places we need to make this change too, like receiving other events from the server? Can you audit for these?